devices/ePaper/Drivers/Ssd168x/Ssd168x.cs

```diff
@@ -627,12 +628,16 @@ protected virtual void WaitMs(int milliseconds)
         }
 
         /// <inheritdoc/>
-        public virtual void WaitReady()
+        public virtual bool WaitReady(int waitingTime)
         {
-            while (_busyPin.Read() == PinValue.High)
+            int currentWait = 0;
+            while (currentWait < waitingTime && _busyPin.Read() == PinValue.High)
```

Comment:
create a cancellation token, it's done for that. And you can use it in the loop as well to be non blocking

Reply:
Something like (just did it from memory) this is much cleaner and you won't have to wait an extra 5 ms (which is ultimately going to be 20 ms) unnecessary. this shows as well how to set a default time.

```csharp
public virtual bool WaitReady(CancellationToken waitingTime = default)
{
  if (waitingTime == default)
  {
    // With a default time of 5 seconds as an example
    waitingTime = new CancellationTokenSource(5000);
  }

  while(!waitingTime.IsCancelledRequest && _busyPin.Read() == PinValue.High)
  {
    waitingTime.WaitHandle.WaitOne(5, true);
  }

  return !waitingTime.IsCancelledRequest;
}
```